includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -655,6 +655,10 @@ public function process_redirect_payment( $order_id, $intent_id, $save_payment_m
 		} catch ( Exception $e ) {
 			Logger::log( 'Error: ' . $e->getMessage() );
 
+			// Confirm our needed variables are set before using them due to there could be a server issue during the get_intent process.
+			$status    = $status ?? null;
+			$charge_id = $charge_id ?? null;
+
 			/* translators: localized exception message */
 			$message = sprintf( __( 'UPE payment failed: %s', 'woocommerce-payments' ), $e->getMessage() );
 			$this->order_service->mark_payment_failed( $order, $intent_id, $status, $charge_id, $message );
```

Comment:
I think I get what you mean. You think the code should be more like this?

```
			if ( isset( $status ) && isset( $charge_id ) ) {
				/* translators: localized exception message */
				$message = sprintf( __( 'UPE payment failed: %s', 'woocommerce-payments' ), $e->getMessage() );
				$this->order_service->mark_payment_failed( $order, $intent_id, $status, $charge_id, $message );
			}
```

I think the problem with that is the order can then sit in Pending Payment status forever with no note as to why it is there. Currently if the order fails, it will get the note as to why it failed.

The code I added with the Order Service only introduced the notice due to I assumed that there would always be a `$status` and `$charge_id` set, when they might not actually be set. Setting them to `null` beforehand allows the order to still fail like it did originally. The Order Service uses those variables, if available, to add the intent status to the order meta, and a link to the transaction in the transactions list via the `charge_id`.

Reply:
Ok, I went through the code and passing null doesn't affect important things.